File: Be/property-service/src/main/java/com/property/dto/response/ApartmentResponse.java
package com.property.dto.response;

import java.util.Set;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Builder
public class ApartmentResponse {
    String apartmentId;
    String userId;

    String name;
    String city;
    String address;
    String rule;
    String utility;
    String description;
    Boolean isAvailable;
    String apartmentType;
    Set<AdditionalCostResponse> additionalCostResponses;
    int currentOccupancy;
}

File: Fe/client/src/dto/response/apartmentResponse.ts
export type apartmentResponse = {
	name: string;
	city: string;
	address: string;
	rule: string;
	utility: string;
	description: string;
	isAvailable: boolean;
	apartmentType: string;
	apartmentId: string;
	additionalCostResponses: Array<additionalCostResponsesType>;
	userId: string;
	currentOccupancy: number;
}

export type additionalCostResponsesType = {
	name: string;
	cost: number;
	additionalCostType: string;
	unit: string;
	additionalCostId: string;
}